### PR TITLE
Add windows build

### PR DESCRIPTION
**Pull request by mmcauliffe**
Adds support for ngram on Windows (dependent on https://github.com/conda-forge/openfst-feedstock/pull/27)

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,28 +8,40 @@ package:
...
+    - m2-patch                # [win]
   host:
     - openfst ==1.8.2
+    - dlfcn-win32             # [win]
   run:
     - openfst ==1.8.2
+    - ucrt                    # [win]
+
+
+{% set ngram_binaries = [
```
Comment (kylebgorman) [inline recipe/meta.yaml:33]: Do you want to just test all of them?

**Review by kylebgorman — COMMENTED** [review]
Hi Michael, this is pretty nice stuff. Does this depend on the OpenFst 1.8.2 Windows patch? I assume it does.

Anyways, I have no issue with one-patch stuff that gets Windows working and will approve when it's ready.

File: recipe/meta.yaml (hunk 1 above)
Comment (mmcauliffe) [reply recipe/meta.yaml:33]: Oh yes, that's just an oversight on my part!

**Comment by mmcauliffe** [comment]
> Hi Michael, this is pretty nice stuff. Does this depend on the OpenFst 1.8.2 Windows patch? I assume it does.
> 
> Anyways, I have no issue with one-patch stuff that gets Windows working and will approve when it's ready.

Yep, it does indeed depend on the 1.8.2 windows patch.  Just working out some kinks in Pynini too and then I'll get that one up too.

File: recipe/patches/0001-Add-CMake-files.patch [hunk 2]
```diff
@@ -0,0 +1,922 @@
...
+ mode change 100755 => 100644 src/test/ngraminfo_test.sh
+ mode change 100755 => 100644 src/test/ngrammake_test.sh
+ mode change 100755 => 100644 src/test/ngrammarginalize_test.sh
+ mode change 100755 => 100644 src/test/ngrammerge_test.sh
+ mode change 100755 => 100644 src/test/ngramperplexity_test.sh
+ mode change 100755 => 100644 src/test/ngramprint_test.sh
+ mode change 100755 => 100644 src/test/ngramrand_test.sh
+ mode change 100755 => 100644 src/test/ngramrandgen_test.sh
+ mode change 100755 => 100644 src/test/ngramshrink_test.sh
+ mode change 100755 => 100644 src/test/ngramsymbols_test.sh
```
Comment (h-vetinari) [inline recipe/patches/0001-Add-CMake-files.patch:84]: I suspect that all these mode changes are unintentional (or at least unnecessary)?

Comment (mmcauliffe) [reply recipe/patches/0001-Add-CMake-files.patch]: Good catch! I went through and cleaned up the patch files for this and other GRM builds.

**Pull request merged**